src/Span.h

```diff
@@ -63,6 +63,11 @@ class Span {
              class = std::enable_if_t<std::is_convertible_v<const Data*, T*>>>
     Span(const Container& xs) noexcept : memory_block(xs.data()), num_elements(xs.size()) {}
 
+    // This only works with contiguous iterators like those from std::vector or std::string!
+    template<class Iter, class Data = typename Iter::value_type,
+             class = std::enable_if_t<std::is_convertible_v<const Data*, T*>>>
+    Span(Iter first, Iter last) noexcept : memory_block(&first[0]), num_elements(last - first) {}
+
```

Comment:
@Neverlord - I suspect this isn't exactly safe or great. If you have some enable_if magic to check for `Iter == std::vector::const_iterator` or some such, that might prevent some trouble.

Can also change the users to use the pointers, but that feels bit clunkier than the iterators.

Reply:
We already have to overload above that takes a container and queries `data()` and `size()`. Why aren't we just using that? I suppose it's because `@ARGS@` in a BIF only has iterator access? But why not extend the interface so that we can do something like `@ARGS@->asSpan()`? I think that would avoid the iterator clunkiness.

Reply:
If it's just a vector, why not `ArgsSpan{*@ARGS@}`?

Reply:
Maybe to clarify, I don't suggest to not use iterators at all, but I would cut this constructor and refactor code places like this:

```
auto ev = zeek::broker_mgr->MakeEvent({@ARGS@->begin(), @ARGS@->end()});
```

to simply

```
auto ev = zeek::broker_mgr->MakeEvent({*@ARGS@->begin()});
```